R/zchunk_L124.nonco2_unmgd_R_S_T_Y.R

```diff
@@ -43,80 +41,186 @@ module_emissions_L124.nonco2_unmgd_R_S_T_Y_DISABLED <- function(command, ...) {
-    # NOTE: This code uses repeat_and_add_vector
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # ===================================================
+    # Reformat temporary data -- Note: this can be removed after the upstream files are done
+    L124.LC_bm2_R_Grass_Yh_GLU_adj %>%
+      gather(year, value, -GCAM_region_ID, -Land_Type, -GLU) %>%
+      mutate(year = as.integer(substr(year, 2, 5))) ->
+      L124.LC_bm2_R_Grass_Yh_GLU_adj
+
+    L124.LC_bm2_R_UnMgdFor_Yh_GLU_adj %>%
+      gather(year, value, -GCAM_region_ID, -Land_Type, -GLU) %>%
+      mutate(year = as.integer(substr(year, 2, 5))) ->
+      L124.LC_bm2_R_UnMgdFor_Yh_GLU_adj
+
+    # Prepare EDGAR emissions for use
+    # Map in region ID and sector name; extrapolate to 2010; aggregate to sector & region; convert from Gg to Tg
+    EDGAR_gases %>%
+      left_join(EDGAR_sector, by = "IPCC") %>%                                          # Add GCAM sector from the sector mapping
+      rename(sector = agg_sector) %>%
+      filter(sector %in% c("grassland", "forest")) %>%                                  # Filter for the two sectors we use in this file.
+      standardize_iso(col = "ISO_A3") %>%
+      change_iso_code('rou', 'rom') %>%                                                 # Switch Romania iso code to its pre-2002 value
+      left_join(iso_GCAM_regID, by = "iso") %>%                                         # Map in GCAM regions
+      select(IPCC, Non.CO2, sector, iso, GCAM_region_ID, year, value) ->
+      EDGAR_history
+
+    if(OLD_DATA_SYSTEM_BEHAVIOR) {
+      # The old data system processed data in wide format. There was a na.omit() in the code that effectively removed
+      # any region/sector combination where the time series was incomplete. We'd like to keep this information.
+      # Additionally, the old data system never included NH3 in the processing. The emissions are small
+      # but we should keep them.
+      EDGAR_history %>%
+        filter(Non.CO2 != "NH3" ) %>%                                                    # Remove NH3 for consistency with old data
+        filter(year <= 2008) %>%                                                         # Old data didn't care if post-2008 data was missing so remove it here
+        spread(year, value) %>%                                                          # Convert to wide format
+        na.omit() %>%                                                                    # Remove any row with an NA (i.e., incomplete time series)
+        gather(year, value, -GCAM_region_ID, -iso, -sector, -Non.CO2, -IPCC) %>%         # Convert back to long format
+        mutate(year = as.integer( year )) ->                                             # Convert year back to integer form (not sure why this changes type)
+        EDGAR_history
+
+    } else {
+      #   ... new code with a fix
```

Comment:
Actually, if the intent is to leave the fix for later, perhaps we should put a `stop` statement in there to give a meaningful error message if we try to run with backward compatibility disabled.  As it stands, with compatibility off this will produce some junk results that might be hard to track down.